docker/Dockerfile

```diff
@@ -1,6 +1,6 @@
 # default to latest supported policy, x86_64
-ARG BASEIMAGE=amd64/almalinux:8
-ARG POLICY=manylinux_2_28
+ARG BASEIMAGE=amd64/almalinux:9
```

Comment:
I was under the impression from [discussion](https://github.com/pypa/manylinux/issues/1282#issuecomment-1548916560) on previous issues that manylinux_2_34 would (might?) use the UBI or UBI-minimal image, though I don't think there was ever an actual conclusion reached on the subject.

https://catalog.redhat.com/software/containers/ubi9/ubi/615bcf606feffc5384e8452e?architecture=amd64&image=65a82982a10f3e68777870b5

https://catalog.redhat.com/software/containers/ubi9/ubi-minimal/615bd9b4075b022acc111bf5?architecture=amd64&image=65a828e3cda4984705d45d26

@carlwgeorge @mattip @tiran @h-vetinari @kpfleming You were all active on the previous discussion, any input?

Reply:
I would love to see manylinux_2_34 use UBI 9 as a base.  UBI is a subset of RHEL packages that are freely redistributable, and RHEL is what Alma aspires to be compatible with.  It's better to just go straight to the primary thing in my opinion.  If there are packages missing from the UBI package set that would block this, we can request those be added (see question 34 [here](https://developers.redhat.com/articles/ubi-faq#support__lifecycle__and_updating-h2)).

Reply:
There really should be a separate discussion issue on `manylinux_2_34`, separate from #1574, to discuss choices/timelines/details around the implementation, with the necessary context, plus referencing the `_2_28` meta-issue and previous discussions ([example](https://discuss.python.org/t/the-next-manylinux-specification/1043/239)) on `_2_34`.

Reply:
> There really needs to be a discussion issue on `manylinux_2_34`

https://github.com/pypa/manylinux/issues/1585

Reply:
tracking issue opened, resolving conversation here.